### PR TITLE
Improvements #1

### LINKED ISSUE #2: Bring types
Add typescript

### PR DESCRIPTION
**Pull request by bogdanbpeterson**
Upgraded to typescript, migrated to PostgreSQL
5fc6550379fffdb5ec5fe227cad5cc13727648e7 resolves #2

### CODE REVIEW

**Comment by bogdanbpeterson** [comment]
ede3c0ee7e7cb5c246c3626c8bf554e3f8647c18 resolves #3

File: src/utils/contains-ruzzniy.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const jobanaruzznia =
```
Comment (sqwoteg) [inline src/utils/contains-ruzzniy.ts:1]: camelCase

File: prisma/schema.prisma [hunk 2]
```diff
@@ -8,6 +11,7 @@ datasource db {
 }
 
 model Chat {
-  id      BigInt @id
-  counter Int    @default(0)
+  chatId Int @id @unique
+
+  encounters Int @default(0)
```
Comment (sqwoteg) [inline prisma/schema.prisma:16]: why did you rename second prop?

File: prisma/schema.prisma [hunk 3]
```diff
@@ -8,6 +11,7 @@ datasource db {
 }
 
 model Chat {
-  id      BigInt @id
-  counter Int    @default(0)
+  chatId Int @id @unique
```
Comment (sqwoteg) [inline prisma/schema.prisma:14]: You should use BigInt there or it will break in some chats

File: src/index.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
+import { Telegraf } from 'telegraf';
+import { message } from 'telegraf/filters';
+import { BOT_TOKEN } from './config/env.config';
+import { jobanaruzzniaFileId } from './config/jobana-ruzznia-animation';
```
Comment (sqwoteg) [inline src/index.ts:4]: jobanaAnimationFileId
jobanaAnimation

File: src/index.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+import { dbClient } from './db';
+import { containsruzzniu } from './utils/contains-ruzzniy';
+import * as chat from './models/chat.model';
+import { pluralize } from './utils/pluralize';
+
+const main = async () => {
+  const bot = new Telegraf(BOT_TOKEN);
+
+  bot.start((ctx) => {
+    ctx.reply('Додай мене в групу і почніть їбати русню');
```
Comment (sqwoteg) [inline src/index.ts:14]: Add different text if it is a group

File: src/index.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  });
+
+  bot.help((ctx) => {
+    // handle help command
```
Comment (sqwoteg) [inline src/index.ts:18]: Useless comment

File: src/index.ts [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    ctx.reply('Додай мене в групу і почніть їбати русню');
```
Comment (sqwoteg) [inline src/index.ts:19]: Add different text if it is a group

File: src/index.ts [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+  });
+
+  bot.on(message('text'), async (ctx) => {
+    if (ctx.message.chat.type === 'private') {
+      ctx.reply('Пробач, але я працюю тільки в групах 🥺');
```
Comment (sqwoteg) [inline src/index.ts:24]: But why?

File: src/index.ts [hunk 9]
```diff
@@ -0,0 +1,64 @@
+import { Telegraf } from 'telegraf';
+import { message } from 'telegraf/filters';
+import { BOT_TOKEN } from './config/env.config';
+import { jobanaruzzniaFileId } from './config/jobana-ruzznia-animation';
+import { dbClient } from './db';
+import { containsruzzniu } from './utils/contains-ruzzniy';
```
Comment (sqwoteg) [inline src/index.ts:6]: Make it a default export

Comment (sqwoteg) [reply src/index.ts:6]: Also, what do you think about renaming it to "contains🐷🐶"?

File: src/index.ts [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    const shouldIAnswer = containsruzzniu(ctx.message.text);
+
+    if (shouldIAnswer) {
+      const chatId = ctx.message.chat.id;
+      const encounters = await chat.incrementCounter(chatId);
+      const timesPluralized = pluralize(encounters, ['раз', 'рази', 'разів']);
+
+      if (encounters === 1) {
+        ctx.reply(
+          `Вітаю! Ваша подорож русофобією розпочалася! ₚусня йобана ${encounters} ${timesPluralized}`
```
Comment (sqwoteg) [inline src/index.ts:37]: Add a linebreak between sentenses

File: src/utils/contains-ruzzniy.ts (hunk 1 above)
Comment (sqwoteg) [reply src/utils/contains-ruzzniy.ts:1]: maybe jobanaExpression will fit better cause you will not write rusnia with a big "R"

**Review by sqwoteg — CHANGES_REQUESTED** [review]

File: src/utils/contains-ruzzniy.ts (hunk 1 above)
Comment (bogdanbryzh) [reply src/utils/contains-ruzzniy.ts:1]: oky

File: prisma/schema.prisma (hunk 2 above)
Comment (bogdanbryzh) [reply prisma/schema.prisma:16]: it's more semantically correct

File: prisma/schema.prisma (hunk 3 above)
Comment (bogdanbryzh) [reply prisma/schema.prisma:14]: will fix, oky

File: src/index.ts (hunk 5 above)
Comment (bogdanbryzh) [reply src/index.ts:14]: makes sense

File: src/index.ts (hunk 8 above)
Comment (bogdanbryzh) [reply src/index.ts:24]: it doesn't make any sense to do it alone 
we have to encourage people to unite on this

File: src/index.ts (hunk 9 above)
Comment (bogdanbryzh) [reply src/index.ts:6]: i prefer not to use default exports unless it's unavoidable

Comment (sqwoteg) [reply src/index.ts:6]: Or if not, rename it co containsJobana

File: src/config/jobanaAnimation.ts [hunk 11]
```diff
@@ -0,0 +1,2 @@
+export const jobanaruzzniaFileId =
```
Comment (sqwoteg) [inline src/config/jobanaAnimation.ts:1]: jobanaFileId

camelCase pls

File: src/utils/containsPiggoDoggos.ts [hunk 12]
```diff
@@ -0,0 +1,6 @@
+const jobanaRegex =
+  /йобан([а-яієї]{1,2}).*(р|ₚ|\.)(у|о)сн([а-яієї]{1,2})|(р|ₚ|\.)(у|о)сн([а-яієї]{1,2}).*йобан([а-яієї]{1,2})/;
+
+export const containsPiggoDogges = (text: string) => {
```
Comment (sqwoteg) [inline src/utils/containsPiggoDoggos.ts:4]: I meant real emojis

If no, lets name it containsJobana

File: src/utils/isChatPrivate.ts [hunk 13]
```diff
@@ -0,0 +1,3 @@
+import type { Chat as TelegrafChat } from 'telegraf/typings/core/types/typegram';
+
+export const isChatPrivate = (chat: TelegrafChat) => chat.type === 'private';
```
Comment (sqwoteg) [inline src/utils/isChatPrivate.ts:3]: maybe unite isChatGroup and isChatPrivate into one file?

File: src/utils/containsPiggoDoggos.ts (hunk 12 above)
Comment (bogdanbpeterson) [reply src/utils/containsPiggoDoggos.ts:4]: emojis doesn't work, will change to `Jobana`

File: src/utils/isChatPrivate.ts (hunk 13 above)
Comment (sqwoteg) [reply src/utils/isChatPrivate.ts:3]: chatUtils.ts maybe

Comment (bogdanbpeterson) [reply src/utils/isChatPrivate.ts:3]: I'd insist on current state, I prefer each function in it's own file. Sometimes might be cluttering, but `ctrl + p` thank you

Comment (sqwoteg) [reply src/utils/isChatPrivate.ts:3]: ok

**Review by sqwoteg — APPROVED** [review]

**Pull request merged**